CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+cmake_minimum_required(VERSION 3.16)
+project(cre2)
+
+# Abseil requires C++14
+set(CMAKE_CXX_STANDARD 14)
+
+add_library(cre2 SHARED "${CMAKE_SOURCE_DIR}/thirdparty/cre2/src/cre2.cpp")
+
+# Define compile definitions that were previously passed via cl.exe.
+target_compile_definitions(cre2 PRIVATE 
+    CRE2_VERSION_INTERFACE_CURRENT=0
+    CRE2_VERSION_INTERFACE_REVISION=0
+    CRE2_VERSION_INTERFACE_AGE=0
+    CRE2_VERSION_INTERFACE_STRING="\"0.0.0\""
+    CRE2_DECL=__declspec(dllexport)
+)
+
+# Add include directories for re2 and Abseil.
+target_include_directories(cre2 PRIVATE 
+    "${CMAKE_SOURCE_DIR}/thirdparty/re2"
+    "C:/vcpkg/installed/x64-windows/include"
+)
+
+# Import Abseil's CMake targets.
+find_package(absl REQUIRED)
+
+# Create an IMPORTED target for re2.
+add_library(re2 STATIC IMPORTED)
+set_target_properties(re2 PROPERTIES
+    IMPORTED_LOCATION "${CMAKE_SOURCE_DIR}/bin/re2/Release/re2.lib"
+)
+
+target_link_libraries(cre2 PRIVATE re2)
```

Comment:
we'll need a target_link_libraries for abseil too.

Reply:
Fine now I think